assets/javascripts/overview.js

```diff
@@ -46,6 +46,17 @@ function setupOverview() {
                 icon.fadeTo('slow', 0.5).fadeTo('slow', 1.0);
             });
         });
+    $('.dependency').each(function() {
+        var deps = JSON.parse($(this).attr('data'));
+        var jobid = $(this).attr('jobid');
+        var result = $(this).attr('result');
+        var dependencyHtml = showJobDependency(deps, jobid);
+        var dependencyResult = showDependencyResult(deps.parents, result);
+        if (dependencyResult !== '') {
+            dependencyResult = '<a href="/tests/' + jobid + '#dependencies">' + dependencyResult + '</a>';
+        }
+        $(this).html(dependencyHtml + dependencyResult);
```

Comment:
I don't like making an HTML string just to call the jQuery `.html` function on it. Instead, just use regular JavaScript functions to create DOM objects directly. (I get that you need `dependencyHtml` because that code is shared with the DataTable which expects HTML. But at least the `dependencyResult` could be inserted via DOM functions.)

I would also generally refrain from using jQuery in new code unless there's a clear benefit because the jQuery library is kind of old fashioned. So better just use the native JavaScript API which is now well supported by most browsers. And if you really have to use jQuery, please refrain from repeated functions calls like `$(this)`.

Reply:
Thanks for your suggestions. I modified the code to not use jQuery. Are these code are 'native Javascript API'?

Reply:
Yes, your new code is using what I call "native JavaScript APIs".